dotcom-rendering/src/web/browser/sentryLoader/loadSentry.ts

```diff
@@ -0,0 +1,66 @@
+import '../webpackPublicPath';
+import { isAdBlockInUse } from '@guardian/commercial-core';
+
+const loadSentry = (): void => {
+	try {
+		// Downloading and initiliasing Sentry is asynchronous so we need a way
+		// to ensure injection only happens once and to capture any other errors that
+		// might happen while this script is loading
+		let injected = false;
+		const queue: Error[] = [];
+
+		// Function that gets called when an error happens before Sentry is ready
+		const injectSentry = async (error?: Error) => {
+			// Remember this error for later
+			if (error) queue.push(error);
+
+			// Only inject once
+			if (injected) {
+				return;
+			}
+			injected = true;
+
+			// Make this call blocking. We are queing errors while we wait for this code to run
+			// so we won't miss any and by waiting here we ensure we will never make calls we
+			// expect to be blocked
+			const adBlockInUse: boolean = await isAdBlockInUse();
+			if (adBlockInUse) {
+				// Ad Blockers prevent calls to Sentry from working so don't try to load the lib
+				return;
+			}
```

Comment:
Should this be extracted to the decision to inject Sentry?

Reply:
I'm just wondering why this wasn't made part of the initial checks?

Is detecting an ad-blocker relatively '[expensive](https://github.com/guardian/commercial-core/blob/3b2a5f35275ef825fa3aa04d8a17a74618de374f/src/detect-ad-blocker.ts#L24)' and therefore we don't want it to block the init path?

`isAdBlockInUse` via  `useAdBlockInUse` is currently only used in islands which don't run on the init path.

Reply:
It's a good point though. I'll add a comment for future travellers.